src/compiler/checker.ts

```diff
@@ -4766,15 +4767,19 @@ namespace ts {
                 }
                 else if (!jsDocType && isBinaryExpression(expression)) {
                     // If we don't have an explicit JSDoc type, get the type from the expression.
-                    let type = getWidenedLiteralType(checkExpressionCached(expression.right));
+                    const isModuleExportsAlias = aliased !== symbol;
+                    let type = isModuleExportsAlias ? getTypeOfSymbol(aliased) : getWidenedLiteralType(checkExpressionCached(expression.right));
 
-                    if (getObjectFlags(type) & ObjectFlags.Anonymous &&
+                    if (type.flags & TypeFlags.Object &&
                         special === SpecialPropertyAssignmentKind.ModuleExports &&
                         symbol.escapedName === InternalSymbolName.ExportEquals) {
-                        const exportedType = resolveStructuredTypeMembers(type as AnonymousType);
+                        const exportedType = resolveStructuredTypeMembers(type as ObjectType);
                         const members = createSymbolTable();
                         copyEntries(exportedType.members, members);
-                        symbol.exports!.forEach((s, name) => {
+                        if (!aliased.exports) {
+                            aliased.exports = createSymbolTable();
+                        }
+                        (isModuleExportsAlias ? aliased : symbol).exports!.forEach((s, name) => {

```

Comment:
If `!isModuleExportsAlias`, `symbol` the same as `aliased`. So this should always equal `aliased`.
Tested with `Debug.assert((isModuleExportsAlias ? aliased : symbol) === aliased);`.

Reply:
This is now `aliased || symbol` now that aliased is optional instead of sometimes duplicated.